### PR TITLE
Using receptor-client from gem

### PR DESCRIPTION
**Pull request by slemrmartin**
Built-in receptor-client was deprecated and has to be replaced with gem to better handle errors and to prevent Segmentation fault crash (see Jira ticket below)

---

**depends on**:
- [x] https://github.com/RedHatInsights/e2e-deploy/pull/2615

---

[RHCLOUD-11474](https://issues.redhat.com/browse/RHCLOUD-11474)

### CODE REVIEW

File: lib/topological_inventory/satellite/connection.rb [hunk 1]
```diff
@@ -1,47 +1,79 @@
...
-                                       :response_callback => :availability_check_response,
-                                       :timeout_callback  => :availability_check_timeout)
+        directive = receptor_client.directive(account_number,
+                                              receptor_node_id,
+                                              :directive => "receptor_satellite:health_check",
+                                              :payload   => {'satellite_instance_id' => source_ref.to_s}.to_json,
+                                              :type      => :non_blocking)
+        directive.on_success { |msg_id, response| receiver.availability_check_response(msg_id, response) }
+                 .on_error { |msg_id, code, response| receiver.availability_check_error(msg_id, code, response) }
+                 .on_timeout { |msg_id| receiver.availability_check_timeout(msg_id) }
```
Comment (lindgrenj6) [inline lib/topological_inventory/satellite/connection.rb:71]: I guess I'm missing something here - where do we call `DirectiveNonBlocking#call` in order for it to actually send the directive?

**Review by lindgrenj6 — COMMENTED** [review]
Just one clarification - looks good otherwise!

File: lib/topological_inventory/satellite/connection.rb (hunk 1 above)
Comment (slemrmartin) [reply lib/topological_inventory/satellite/connection.rb:71]: you should be right, now I don't understand why it worked :)

Comment (slemrmartin) [reply lib/topological_inventory/satellite/connection.rb:71]: fixed, this [super](https://github.com/RedHatInsights/topological_inventory-satellite/pull/44/files#diff-20ab2f2e7ab7747f888230e05de86e7a2fdd3d76a424e273ea29b8a86dd5f8d8L29) returned false available

**Review by lindgrenj6 — APPROVED** [review]

**Pull request merged**